lib/asset_cloud/base.rb

```diff
@@ -8,8 +8,25 @@ class IllegalPath < StandardError
 
   class Base
     cattr_accessor :logger
-
-    VALID_PATHS = /\A[a-z0-9][a-z0-9_\-\/]+([a-z0-9][\w\-\ \.\@]*\.\w{2,6})?\z/i
+    VALID_PATHS = /\A
+      (
+        ([a-z0-9])            #Filename can be a single letter
+        |                     #OR it is many and follows the below rules
+        (
+          (\.?[\w\-\@])       #It can start with a dot but it must have a following character
+          (
+            [\w\-\@]          #You can have a letter without any following conditions
+            |
+            [\ ][\w\-\@]      #If there is a space you need to have a normal letter afterward
+            |
+            [\/][\w\-\@]      #If there is a slash you need to have a normal letter afterward
+            |
+            [\/][\.][\w\-\@]  #Though a slash could be followed by a dot so long as there is a normal letter afterward
+            |
+            [\.][\w\-\@]+     #A dot must be followed by one (or more) normal letters
+          )*                  #Zero to many of these combinations.
+        )
+      )\z/ix                   #Case insensitive
```

Comment:
I know you meant different regexes, but I suspect that would simply add complexity. 

A wise man once said, in this very PR:
> Some people, when confronted with a problem, think 
> “I know, I'll use regular expressions.” Now they have two problems.

I'm thinking:
> Some people confronted with a regex, think
> "I know, I'll use 3 regexes", Now they have 4 problems.